rust/src/tx_builder.rs

```diff
@@ -638,29 +655,66 @@ impl TransactionBuilder {
 
     /// Set explicit Mint object to this builder
     /// it will replace any previously existing mint
+    /// NOTE! If you use `set_mint` manually - you must use `set_mint_scripts`
+    /// to provide matching policy scripts or min-fee calculation will be rejected!
```

Comment:
Is it possible to enforce this is set if you set the mint in the build function and return an error otherwise?

Reply:
🤔 Might be a good idea. Need to think a bit, because it's inconsequential for the original `build()` as it only returns the tx body. Might be added into the `build_tx()`, but on the other hand the only place where it actually breaks the logic is the fee estimation as we can guarantee that the estimate will be wrong without the scripts being present. That's why I have already added the assertion and an error into the `min_fee` function in case matching scripts are not found.

Reply:
@rooooooooob , what do you think of combining the `set_min` and `set_mint_scripts` into a single function like

```
/// Set explicit Mint object and witness mint NativeScripts to this builder
/// it will replace any previously existing mint and witness scripts
/// NOTE! The scripts must contain all entries to match policy hashes used in the mint
/// If no matching script is found for a policy - error will be returned and the builder state will not change
pub fn set_mint(&mut self, mint: &Mint, mint_scripts: &NativeScripts) -> Result<(), JsError> { ... }
```

Reply:
That sounds like an even better idea since it's simpler and they're gonna need it to make the full tx later anyway.

Reply:
Will add the change for this then